### PR TITLE
Fixing errors in e2e test

### PR DESCRIPTION
**Pull request by ace2107**
Removed version hardcoding and Changes made in test and feature file

### CODE REVIEW

File: features/analyse_container_image.feature [hunk 1]
```diff
@@ -9,6 +9,8 @@ Feature: Analyse Container Image
         Then I want to receive a Analyser Job ID
         And I wait for the Analyser Job to finish successful
         And the Analyzer Job Log should not be empty
+		Then I query the Result API for the result of the latest analyzer
```
Comment (fridex) [inline features/analyse_container_image.feature]: Indentation.

File: features/steps/result_api_steps.py [hunk 2]
```diff
@@ -88,6 +88,17 @@ def step_impl(context):
...
 
 
+@then('I query the Result API for the result of the latest analyzer')
+def step_impl(context):
+    r = requests.get(f'{context.endpoint_url}/analyze/{context.latest_doc_id}')
+
+    assert_that(r.status_code, equal_to(HTTPStatus.OK))
+    assert_that(r.headers['content-type'], equal_to('application/json'))
+    analyzer_result_json = r.json()
+    assert_that(analyzer_result_json,not_none)
```
Comment (fridex) [inline features/steps/result_api_steps.py]: Missing space after ",". Here and in other places as well.

File: features/steps/result_api_steps.py [hunk 3]
```diff
@@ -104,10 +115,10 @@ def step_impl(context, name):
                 equal_to(name))
 
 
-@then('the analyser version should be "{version}"')
-def step_impl(context, version):
+@then('the analyser version should not be empty')
+def step_impl(context):
     assert_that(context.chosen_result_json['metadata']['analyzer_version'],
-                equal_to(version))
+                not equal_to(0))
```
Comment (fridex) [inline features/steps/result_api_steps.py]: You are checking it here against zero, but the step says it should not be empty. Here and the solver case as well.

File: features/steps/result_api_steps.py [hunk 4]
```diff
@@ -116,7 +127,13 @@ def step_impl(context, name):
...
 
 
-@then(u'the solver version should be "{version}"')
-def step_impl(context, version):
+@then(u'the solver version should not be empty')
+def step_impl(context):
     assert_that(context.chosen_result_json['metadata']['analyzer_version'],
-                equal_to(version))
+                not equal_to(0))
+    print('Solver version number = ', context.chosen_result_json['metadata']['analyzer_version'])
```
Comment (fridex) [inline features/steps/result_api_steps.py]: Please avoid having print statements in the source code.

File: features/steps/result_api_steps.py [hunk 5]
```diff
@@ -116,7 +127,13 @@ def step_impl(context, name):
...
+
+@then(u'the list of analzer results should not be empty')
+def step_impl(context):
+    assert_that(len(context.latest_analyzer_result['metadata']),not equal_to(0))
```
Comment (fridex) [inline features/steps/result_api_steps.py]: You are checking the metadata part. Why?

File: features/steps/user_api_steps.py [hunk 6]
```diff
@@ -24,6 +24,8 @@
 
 import requests
 
+import re
```
Comment (fridex) [inline features/steps/user_api_steps.py]: Unused import.

File: features/analyse_container_image.feature [hunk 7]
```diff
@@ -18,10 +20,3 @@ Feature: Analyse Container Image
     Scenario: Query for currently available Analyser Results
         When I query the Result API for a list of analyser results
         Then the list of results should not be empty
-
```
Comment (goern) [inline features/analyse_container_image.feature]: why has this scenario been removed?!

Comment (ace2107) [reply features/analyse_container_image.feature]: Two scenarios namely "Submitting a Container Image for analysis" and  "Get the Analyser Results for my last submitted Container Image" from analyse_container_image.feature are correlated and one main scenario . Only once first scenario is passed , we can query the result API for details of the latest analyzer result. So it hasnt been removed but 2 of them have been merged

File: features/steps/result_api_steps.py [hunk 8]
```diff
@@ -116,7 +126,12 @@ def step_impl(context, name):
                 equal_to(name))
 
 
-@then(u'the solver version should be "{version}"')
-def step_impl(context, version):
-    assert_that(context.chosen_result_json['metadata']['analyzer_version'],
-                equal_to(version))
+@then(u'the solver version should not be empty')
```
Comment (fridex) [inline features/steps/result_api_steps.py]: Why do you put `u` in front of strings?

File: features/steps/user_api_steps.py [hunk 9]
```diff
@@ -47,20 +47,16 @@ def step_impl(context, a_container_image, analyser_image):
 
 @when(u'I query the Result API for my latest analyser result')
 def step_impl(context):
-    payload = {
-        'pod_id': context.analyzer_job_id
-    }
-    r = requests.get(
-        f'{context.endpoint_url}/log', params=payload)
+    analyzer_job_id = context.response['pod_id']
+    r = requests.get(f'{context.endpoint_url}/log/{analyzer_job_id}')
```
Comment (fridex) [inline features/steps/user_api_steps.py]: Why are you hitting the `/log` endpoint here?

File: features/steps/user_api_steps.py [hunk 10]
```diff
@@ -47,20 +47,16 @@ def step_impl(context, a_container_image, analyser_image):
...
 
     assert_that(r.status_code, equal_to(HTTPStatus.OK))
     assert_that(r.headers['content-type'], equal_to('application/json'))
-
+
```
Comment (fridex) [inline features/steps/user_api_steps.py]: Whitespace not needed.

File: features/steps/user_api_steps.py [hunk 11]
```diff
@@ -110,13 +105,19 @@ def step_impl(context):
...
 
     analyzer_job_log = r.json()
     assert_that(analyzer_job_log, not_none)
+    log_data = r.json()
+    assert_that(log_data, not_none)
+    document_id = log_data['pod_id']
+    assert_that(document_id, not_none)
 
-    assert_that(analyzer_job_log['pod_id'], equal_to(context.analyzer_job_id))
+    context.latest_doc_id = document_id
```
Comment (fridex) [inline features/steps/user_api_steps.py]: Why are you assigning here? The step itself does not mention you are changing in the context.

File: features/analyse_container_image.feature (hunk 7 above)
Comment (goern) [reply features/analyse_container_image.feature]: duh, that was not obvious form the PR description :)

**Comment by goern** [comment]
/lgtm
ptal @fridex

File: features/analyse_container_image.feature [hunk 12]
```diff
@@ -9,6 +9,8 @@ Feature: Analyse Container Image
         Then I want to receive a Analyser Job ID
         And I wait for the Analyser Job to finish successful
         And the Analyzer Job Log should not be empty
+        Then I query the Result API for the result of the latest analyzer
+        And the list of analzer results should not be empty
```
Comment (fridex) [inline features/analyse_container_image.feature]: anal**y**zer

File: features/steps/result_api_steps.py [hunk 13]
```diff
@@ -104,19 +116,23 @@ def step_impl(context, name):
...
-    assert_that(context.chosen_result_json['metadata']['analyzer_version'],
-                equal_to(version))
+@then('the solver version should not be empty')
+def step_impl(context):
+    assert_that(len(context.chosen_result_json['metadata']['analyzer_version']), not equal_to(0))
+
+
+@then('the list of analzer results should not be empty')
+def step_impl(context):
+    assert_that(len(context.latest_analyzer_result['metadata']['analyzer']), not equal_to(0))
```
Comment (fridex) [inline features/steps/result_api_steps.py]: This is confusing - the step states "'the list of analzer results should not be empty" but AFAIS it checks that the name of analyzer has non-zero length.

File: features/steps/user_api_steps.py [hunk 14]
```diff
@@ -105,18 +82,14 @@ def step_impl(context):
...
-        f'{context.endpoint_url}/log', params=payload)
+        f'{context.endpoint_url}/log/{analyzer_job_id}')
 
     assert_that(r.status_code, equal_to(HTTPStatus.OK))
     assert_that(r.headers['content-type'], equal_to('application/json'))
 
     analyzer_job_log = r.json()
     assert_that(analyzer_job_log, not_none)
-
-    assert_that(analyzer_job_log['pod_id'], equal_to(context.analyzer_job_id))
```
Comment (fridex) [inline features/steps/user_api_steps.py]: This should be always true, but you can keep it just to be sure :)

File: features/analyse_container_image.feature (hunk 12 above)
Comment (goern) [reply features/analyse_container_image.feature]: @ace2107 ?!

File: features/steps/result_api_steps.py (hunk 13 above)
Comment (goern) [reply features/steps/result_api_steps.py]: @ace2107 ?!

**Comment by goern** [comment]
@ace2107 please have a look at this again, its open for way too long!

**Pull request merged**